### DISCUSSION TITLE
DELETE Code Template Library & API Review

### DISCUSSION THREAD

**Discussion by boukenka** [nextgenhealthcare/connect#5124] [Feature/Enhancement Ideas]
Hello,

Today, to delete a code template library, you must first retrieve the XML content of the code template libraries.
Then delete the section of the code template library that you do not want to keep.
Finally to do a bulk update. This is a complex and resource intensive process.
Why not implement a DELETE /codeTemplateLibraries/{libraryId} API ?
There are many operations with the APIs that are complex and don't match typical REST APIs.
They should be updated without multipart/form and work with JSON by default.
These new APIs could be called with /api/v2/... so there will be no breaking change.
Could this be planned in the future?

**Comment by pacmano1** [comment]
Do check the current issues prior to posting.  For example please read https://github.com/nextgenhealthcare/connect/issues/4549, which I believe is your first request.

Perhaps further define "are complex".   And JSON is supported for all calls, it's a simple option to add to your call and changing that behavior to "default" might break prior code for other users.

**Reply by boukenka** [reply]
> Do check the current issues prior to posting. For example please read #4549, which I believe is your first request.

I have encountered an issue regarding the implementation of a specific API. The official support replied me to create an issue in the Github discussion section to report this idea. I didn't know that it had already been reported.

> Perhaps further define "are complex". 

Complex means that it should not be necessary to use multipart/form, to use application/x-www-form-urlencoded and to load/parse the XML before calling an API.  There is an example in my previous post. 😄 When I look at the date for the issue, I have to say that this was reported nearly a year ago. No change since then...

> And JSON is supported for all calls, it's a simple option to add to your call and changing that behavior to "default" might break prior code for other users.

JSON did not work for all API calls when I was testing them before.  I had to switch to XML for some of them to get it to work. So I'm happy that now all APIs work successfully with JSON.

I hope in the near future, there will be some progress on that front.

**Reply by boukenka** [reply]
Thank you for taking the time to answer me.

**Comment by siddharthsharma9258** [comment]
I am not sure if exposing code templates through RESTful APIs should be an option at all. Its all your useful and reusable code and by design shareable across multiple channels, and groups. 
Implementing this  DELETE /codeTemplateLibraries/{libraryId} API is very dangerous.

**Reply by boukenka** [reply]
It is not this option that is dangerous. It is dangerous if any user can access the functionality without confirmation. Moreover, it is already possible to do this via the API in a complicated way.

**Reply by pacmano1** [reply]
I tend to agree with @boukenka here, that normal CRUD operations should be supported in the API.    It is up to the person using the API to make sure they are used appropriately.
